### PR TITLE
Adding regression test for indentation and col wrapping

### LINKED ISSUE #510: line break in header leads to incorrect row mapping of indents
When there is line break in header, the indents of the rows are not calculated correctly
`devtools::session_info()` at the end of your post.

```r
library(dplyr)
library(rtables)
ANL <- DM %>% mutate(value = rnorm(n()), pctdiff = runif(n()))

## toy example where we take the mean of the first variable and the
## count of >.5 for the second.
colfuns <- list(function(x) in_rows(mean = mean(x), .formats = "xx.x"),
                function(x) in_rows("# x > 5" = sum(x > .5), .formats = "xx"))

basic_table() %>%
    split_cols_by("ARM") %>%
    split_cols_by_multivar(c("value", "pctdiff"), varlabels = c("abc", "def")) %>%
    split_rows_by("RACE", split_label = "ethnicity",
                  split_fun = drop_split_levels) %>%
    summarize_row_groups() %>%
    analyze_colvars(afun = colfuns) %>%
    build_table(ANL)

basic_table() %>%
    split_cols_by("ARM") %>%
    split_cols_by_multivar(c("value", "pctdiff"), varlabels = c("abc", "de\nf")) %>%
    split_rows_by("RACE", split_label = "ethnicity",
                  split_fun = drop_split_levels) %>%
    summarize_row_groups() %>%
    analyze_colvars(afun = colfuns) %>%
    build_table(ANL)
```
image

```
─ Session info ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23)
 os       Ubuntu 20.04.5 LTS
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Etc/UTC
 date     2023-01-30
 pandoc   2.19.2 @ /usr/local/bin/pandoc

─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version  date (UTC) lib source
 assertthat    0.2.1    2019-03-21 [1] CRAN (R 4.2.1)
 cachem        1.0.6    2021-08-19 [1] CRAN (R 4.2.1)
 callr         3.7.2    2022-08-22 [1] CRAN (R 4.2.1)
 cli           3.4.1    2022-09-23 [1] CRAN (R 4.2.1)
 crayon        1.5.1    2022-03-26 [1] CRAN (R 4.2.1)
 DBI           1.1.3    2022-06-18 [1] CRAN (R 4.2.1)
 devtools      2.4.4    2022-07-20 [1] CRAN (R 4.2.1)
 digest        0.6.30   2022-10-18 [1] CRAN (R 4.2.1)
 dplyr       * 1.0.10   2022-09-01 [1] CRAN (R 4.2.1)
 ellipsis      0.3.2    2021-04-29 [1] CRAN (R 4.2.1)
 fansi         1.0.3    2022-03-24 [1] CRAN (R 4.2.1)
 fastmap       1.1.0    2021-01-25 [1] CRAN (R 4.2.1)
 formatters  * 0.3.4.15 2023-01-30 [1] git2r (https://github.com/insightsengineering/formatters.git@4c07b43)
 fs            1.5.2    2021-12-08 [1] CRAN (R 4.2.1)
 generics      0.1.3    2022-07-05 [1] CRAN (R 4.2.1)
 glue          1.6.2    2022-02-24 [1] CRAN (R 4.2.1)
 htmltools     0.5.3    2022-07-18 [1] CRAN (R 4.2.1)
 htmlwidgets   1.5.4    2021-09-08 [1] CRAN (R 4.2.1)
 httpuv        1.6.6    2022-09-08 [1] CRAN (R 4.2.1)
 jsonlite      1.8.0    2022-02-22 [1] CRAN (R 4.2.1)
 later         1.3.0    2021-08-18 [1] CRAN (R 4.2.1)
 lifecycle     1.0.2    2022-09-09 [1] CRAN (R 4.2.1)
 magrittr    * 2.0.3    2022-03-30 [1] CRAN (R 4.2.1)
 memoise       2.0.1    2021-11-26 [1] CRAN (R 4.2.1)
 mime          0.12     2021-09-28 [1] CRAN (R 4.2.1)
 miniUI        0.1.1.1  2018-05-18 [1] CRAN (R 4.2.1)
 pillar        1.8.1    2022-08-19 [1] CRAN (R 4.2.1)
 pkgbuild      1.3.1    2021-12-20 [1] CRAN (R 4.2.1)
 pkgconfig     2.0.3    2019-09-22 [1] CRAN (R 4.2.1)
 pkgload       1.3.0    2022-06-27 [1] CRAN (R 4.2.1)
 prettyunits   1.1.1    2020-01-24 [1] CRAN (R 4.2.1)
 processx      3.7.0    2022-07-07 [1] CRAN (R 4.2.1)
 profvis       0.3.7    2020-11-02 [1] CRAN (R 4.2.1)
 promises      1.2.0.1  2021-02-11 [1] CRAN (R 4.2.1)
 ps            1.7.1    2022-06-18 [1] CRAN (R 4.2.1)
 purrr         0.3.4    2020-04-17 [1] CRAN (R 4.2.1)
 R6            2.5.1    2021-08-19 [1] CRAN (R 4.2.1)
 Rcpp          1.0.9    2022-07-08 [1] CRAN (R 4.2.1)
 remotes       2.4.2    2021-11-30 [1] CRAN (R 4.2.1)
 rlang         1.0.6    2022-09-24 [1] CRAN (R 4.2.1)
 rtables     * 0.5.3.4  2023-01-30 [1] git2r (https://github.com/Roche/rtables.git@231ffa0)
 sessioninfo   1.2.2    2021-12-06 [1] CRAN (R 4.2.1)
 shiny         1.7.2    2022-07-19 [1] CRAN (R 4.2.1)
 stringi       1.7.8    2022-07-11 [1] CRAN (R 4.2.1)
 stringr       1.4.1    2022-08-20 [1] CRAN (R 4.2.1)
 tibble        3.1.8    2022-07-22 [1] CRAN (R 4.2.1)
 tidyselect    1.1.2    2022-02-21 [1] CRAN (R 4.2.1)
 urlchecker    1.0.1    2021-11-30 [1] CRAN (R 4.2.1)
 usethis       2.1.6    2022-05-25 [1] CRAN (R 4.2.1)
 utf8          1.2.2    2021-07-24 [1] CRAN (R 4.2.1)
 vctrs         0.4.1    2022-04-13 [1] CRAN (R 4.2.1)
 xtable        1.8-4    2019-04-21 [1] CRAN (R 4.2.1)

 [1] /usr/local/lib/R/site-library
 [2] /usr/local/lib/R/library


```

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #510

### CODE REVIEW

File: tests/testthat/test-printing.R [hunk 1]
```diff
@@ -333,7 +333,7 @@ test_that("Inset works for table, ref_footnotes, and main footer", {
 
   expect_false(result[[1]]) # No inset
   expect_true(result[[2]]) # Inset
-  expect_true(all(vec_tt[sep_index + 1] == "   ===================="))
+  expect_true(all(vec_tt[sep_index + 1] == "   ======================"))
```
Comment (Melkiades) [inline tests/testthat/test-printing.R:336]: Why has this changed?

Comment (gmbecker) [reply tests/testthat/test-printing.R:336]: It shouldn't have as far as I know, if the body of the test hasn't changed...

Comment (gmbecker) [reply tests/testthat/test-printing.R:336]: Ah, the body of the test did change. You're (correctly) calling matrix_form(tt, TRUE) now. this makes the table wider, so the divider is wider.

**Comment by Melkiades** [comment]
The error that will come out is due to how the indentation is saved into `mf_rinfo(__)$indent`. If it is only one the current code puts only a single " " back in but the original table has 2?? I guess the default is actually 2 spaces for indent == 1, is it @gmbecker? I am investigating how to put it correctly back to reflect current state of things ;)

**Comment by gmbecker** [comment]
@Melkiades `mf_rinfo()$indent` is the number of indents (in the same sense that indent modifiers modify). This is independent of _indent size_ which is not a property of the table but rather set at the `toString`/`export_as_txt`, etc level via the `indent_size` parameter

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @Melkiades . tested by Liming, in downstream chevron. everything is working now

**Comment by shajoezhu** [comment]
Hi @gmbecker , hope you are well. Can you merge the changes here. The problem is now resolved. and we have several PRs downstream in Chevron is blocked by this. Thanks a lot!

**Pull request merged**